### DISCUSSION TITLE
HOW TO: Trust API Certificate on Mac OS

### DISCUSSION THREAD

**Discussion by JessicaSprague** [coddingtonbear/obsidian-web#209] [General]
It's working! And I wanted to share because I am setting up the API using Chrome on Mac Sonoma and it took awhile (couple hours tbh) to figure out how to approximate the Windows instructions, which are completely different on Mac. I was receiving the error 

<img width="910" alt="341063164-b1af605d-c2ea-4bea-8ab1-76555a649e66" src="https://github.com/user-attachments/assets/bff578cf-07f3-460d-93e2-aa3d7082c4ce">

This was reported as a bug, but isn't. It's just a lack of instructions for adding and configuring the certificate on Mac OS (which Chrome defaults to and can't/won't change, no matter how loudly you shout)

SOLUTION: 

1. Download the certificate from the Settings in the API in Obsidian, and navigate TO the certificate in Finder (don't try to open it w Chrome via the download history) 
2. Double-click the certificate file, and add the certificate to the System Keychain (you'll need to log in with your Mac password)
3. Under Trust, set to Always Trust. 

<img width="1018" alt="ObsidianScreenshot" src="https://github.com/user-attachments/assets/5f8b8ddf-bfcb-4b02-95e7-66edc9a848e8">

Now check the browser extension by right-clicking it in Chrome, and the little red shield of NO! next to your API key should have turned into a friendly green checkmark of YES! and that particular error should be gone. 

I noted that Mac OS aggressively dissuades from adding certificates manually, so it probably seems like you're doing something terrible and world-ending, but then it's like oh, ok that's fine. And it is.

**Comment by coddingtonbear** [comment]
~~I'm afraid I don't have a mac anywhere handy these days, so I won't be able to help lend much insight into the certificate trusting process on OSX, but once you do find a solution I hope you would consider adding some instructions to https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F to cover OSX.~~

Now that I'm actually _reading_ your comment instead of just looking at the screenshots...I suppose you've already done that work.  I'll move these into the Wiki.

Thanks for taking the time to write up a solution for folks -- I'm sure lots of people will find it helpful!

**Comment by coddingtonbear** [comment]
Thanks agaain, @JessicaSprague -- I've added your instructions to the wiki here: https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F#user-content-macos-aka-os-x
